### PR TITLE
[docs]: Move documentation from wiki to docusaurus

### LINKED ISSUE #81: Feat : Moving the documentation to docusaurus
### Is your feature request related to a problem? Please describe.
Current documentation is fine but with docusaurus , the documentation becomes more readable and maintainable

### Describe the solution you'd like
Moving the project's documentation to [docusaurus](https://v2.docusaurus.io/)

### Additional context
Link to the discussion [here ](https://anitab-org.zulipchat.com/#narrow/stream/237907-open-source-progs/topic/Documentation)

### PR DESCRIPTION
**Pull request by Rahulm2310**
### Description

Move documentation from wiki to docusaurus.

Fixes #81 

### Type of Change:

- Code
- Documentation

**Code/Quality Assurance Only**

- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?

Documentation has been hosted to http://osp-web-docs.surge.sh/. 

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have made corresponding changes to the documentation

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by Rahulm2310** [comment]
@isabelcosta Can you please review this PR.

**Review by isabelcosta — CHANGES_REQUESTED** [review]
Could you add a structure to docusaurus similar to the Wiki Sidebar @Rahulm2310 ?
So there is a hierarchy related to the documentation :)

Thank you for working on this!

**Review by Aaishpra — COMMENTED** [review]
Everything looks correct to me, if you can add what @isabelcosta suggested it could be better.

File: docs/docs/Coding-Standards.md
Comment (Aaishpra) [inline docs/docs/Coding-Standards.md]: Can we also add a description about using [Eslint](https://eslint.org/) as a code linter for the project, I saw usage of `Eslint` in #75 issue

**Review by Aaishpra — CHANGES_REQUESTED** [review]
@isabelcosta what do you think about my suggestion. 🤔

File: docs/README.md
Comment (ashwinijha6) [inline docs/README.md]: @Rahulm2310  I guess there is no use of the word console here you can simply use ``` without console. And wherever mentioned

**Review by ashwinijha6 — CHANGES_REQUESTED** [review]

File: docs/README.md
Comment (ashwinijha6) [inline docs/README.md]: In this or & You can deploy do not appear on different lines in the preview
doc1
you could just mention Or, you can deploy . . .

File: docs/README.md
Comment (ashwinijha6) [inline docs/README.md]: just put a space after line & before (happens only once).

**Review by ashwinijha6 — CHANGES_REQUESTED** [review]

File: docs/docs/Home.md
Comment (ashwinijha6) [inline docs/docs/Home.md]: Change this to Welcome to the Open Source Programs Web rather than wiki.

**Review by ashwinijha6 — CHANGES_REQUESTED** [review]

File: docs/docs/Fork,-Clone,-Remote-and-Pull-Request.md
Comment (Amulya-coder) [inline docs/docs/Fork,-Clone,-Remote-and-Pull-Request.md]:
```suggestion
**Note:** _For more detailed understanding visit [this page](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/overview)_.
```
remove underscore after the full stop

**Review by Amulya-coder — CHANGES_REQUESTED** [review]

File: docs/docs/Getting-Started.md
Comment (Amulya-coder) [inline docs/docs/Getting-Started.md]:
```suggestion
- Once you get the approval you can start working on it.
```
 Please add full-stop after the ending of each line.

**Review by Amulya-coder — CHANGES_REQUESTED** [review]

File: docs/docs/Maintainer-Guidelines.md
Comment (Amulya-coder) [inline docs/docs/Maintainer-Guidelines.md]:
```suggestion
## Code Review

- When possible get a second opinion before merging.
- Make sure the PR has a link to the issue.
- If you can, let them know why your change request makes sense.
```
Add space here!

File: docs/docs/Coding-Standards.md
Comment (Rahulm2310) [reply docs/docs/Coding-Standards.md]: @Aaishpra As there is no PR for #75 , this can be added later through a first timer issue. What say ?

**Comment by Rahulm2310** [comment]
@isabelcosta Applied suggested changes. Is it okay now?

File: docs/docs/Getting-Started.md
Comment (ashwinijha6) [inline docs/docs/Getting-Started.md]: @Rahulm2310 you could remove the exclamation from the end of the last point 10

**Review by ashwinijha6 — CHANGES_REQUESTED** [review]

File: docs/docs/Future-Ideas.md
Comment (Amulya-coder) [inline docs/docs/Future-Ideas.md]:
```suggestion
If you want to implement one of these ideas feel free to start a discussion about this on AnitaB.org Open Source Zulip [#**open-source-progs>New Issues**](https://anitab-org.zulipchat.com/#narrow/stream/237907-open-source-progs/topic/New.20Issues).
```
@Rahulm2310 please remove the exclamatory mark from here also it's not needed.

File: docs/docs/Getting-Started.md
Comment (ashwinijha6) [inline docs/docs/Getting-Started.md]: @Rahulm2310 for the above line you should write Docusaurus instead of Github Wiki for the getting started with the project, Also provide the relevant link that is for web & also the correct the link for the README it should be the web README.
Or if it is like getting started with the project requires setting up the backend first you could mention a line separately for that. What say?

**Review by ashwinijha6 — CHANGES_REQUESTED** [review]

File: docs/docs/Getting-Started.md
Comment (ashwinijha6) [reply docs/docs/Getting-Started.md]: Also you could mention a line related to the video setup

File: docs/docs/Coding-Standards.md
Comment (Aaishpra) [reply docs/docs/Coding-Standards.md]: I understand your point about first-timer issues but we don't want all the issues to be documentation related for the first-timers (considering the point we already have a lot of documentation related issues in this project when even the MVP is not complete) and since there is no PR for #75 how come we are adding usage of Prettier in the wiki (there is no PR for prettier too if not for Eslint)

File: docs/docs/2020-Bismita-Guha.md
Comment (ashwinijha6) [inline docs/docs/2020-Bismita-Guha.md]: @Rahulm2310  just a small change the spelling of Speed is wrong in the last line. Could you pls correct it

**Review by ashwinijha6 — CHANGES_REQUESTED** [review]

File: docs/docs/2020-Bismita-Guha.md
Comment (ashwinijha6) [reply docs/docs/2020-Bismita-Guha.md]: 5th point

File: docs/docs/Coding-Standards.md
Comment (Rahulm2310) [reply docs/docs/Coding-Standards.md]: >but we don't want all the issues to be documentation related for the first-timers (considering the point we already have a lot of documentation related issues in this project when even the MVP is not complete)

@Aaishpra documentation issues are in no way blocking the completion of MVP. So I don't think we need to worry about the number of documentation issues rising. 

>how come we are adding usage of Prettier in the wiki (there is no PR for prettier too if not for Eslint)

Even I don't know why it is there in the wiki. @isabelcosta Can you please give your input here.

Comment (isabelcosta) [reply docs/docs/Coding-Standards.md]: Feel free to remove that if you think that Prettier is not being used. I would also suggest avoiding changing too much in this PR, and make this only about moving the wiki to the repository and leave the documentation fixes to another PR.

Comment (isabelcosta) [reply docs/docs/Coding-Standards.md]: Documentation might have been written initially with a plan for the project in mind, and then the plan did not get executed. So feel free to change this line to a more accurate version of the project.

Comment (Rahulm2310) [reply docs/docs/Coding-Standards.md]: Thanks @isabelcosta 🙂

**Review by isabelcosta — DISMISSED** [review]
I approve, assuming this is as is on the Wiki!
However, please take into consideration other reviews :) Once this has 2 approvals, I can merge.

Thank you for doing this @Rahulm2310 🚀

File: docs/src/pages/index.js
Comment (ashwinijha6) [inline docs/src/pages/index.js]: @Rahulm2310 this main tag is not required here

**Review by ashwinijha6 — CHANGES_REQUESTED** [review]

**Comment by Rahulm2310** [comment]
@ashwinijha6 I would be grateful if you could review all changes together. 🙂

**Comment by ashwinijha6** [comment]
> @ashwinijha6 I would be grateful if you could review all changes together. 🙂

I'm so sorry @Rahulm2310 actually I was working on the backend of docusaurus & just saw this.

**Comment by Rahulm2310** [comment]
@isabelcosta Please add the `Need Review` label. Thanks 🙂

**Review by ashwinijha6 — APPROVED** [review]
Great work @Rahulm2310 🚀
Thanku for making the necessary changes

**Review by isabelcosta — APPROVED** [review]
Great work @Rahulm2310 !
Thank you for your contribution! 🚀

**Pull request merged**
